data/src/user.rs

```diff
@@ -23,17 +23,27 @@ impl Hash for User {
 
 impl Ord for User {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        other
-            .highest_access_level()
-            .cmp(&self.highest_access_level())
+        let access_level_comparison = other.highest_access_level()
+            .cmp(&self.highest_access_level());
+
+        if access_level_comparison != std::cmp::Ordering::Equal {
+            return access_level_comparison;
```

Comment:
You can chain orderings so you don't need to early return. Checkout Ordering::then.

Reply:
I was able to change the code for cmp. But couldn't figure it out for partial_cmp. Like I said this is my first time working in Rust. Can you take a look and let me know if I made the change correctly?

Also, can you suggest how to change partial_cmp?

Reply:
Just call access_level.cmp as well then wrap the returned Ordering in Some

Reply:
I came up with this but I think it becomes way less ledgible than the original:

```
impl PartialOrd for User {
    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
        other.highest_access_level()
            .partial_cmp(&self.highest_access_level())
            .and_then(|access_cmp| {
                other.nickname().partial_cmp(&self.nickname()).map(|nickname_cmp| {
                    access_cmp.then(nickname_cmp)
                })
            })
            
        // Original
        // let access_level_comparison = other.highest_access_level()
        //     .partial_cmp(&self.highest_access_level());
        //
        // if access_level_comparison != Some(std::cmp::Ordering::Equal) {
        //     return access_level_comparison;
        // }
        //
        // other.nickname().partial_cmp(&self.nickname())
    }
}
```

Thoughts?

Reply:
Both AccessLevel and Nick are Eq so we can just use `cmp` instead of partial_cmp.

The body here can be identical to what we used in the Eq body, just wrapping the return value in Some

Reply:
Yes. Spoke with Casper and he clarified. I submitted the change.